base/compiler/abstractlattice.jl

```diff
@@ -131,6 +131,23 @@ remaining mostly associative and commutative.
 """
 function tmerge end
 
+"""
+    tmerge_field(𝕃::AbstractLattice, a, b) -> nothing or lattice element
+
+Compute a lattice join of elements `a` and `b` over the lattice `𝕃`,
+where `a` and `b` are fields of `PartialStruct` or `Const`.
+This is an opt-in interface to allow external lattice implementation to provide its own
+field-merge strategy. If it returns `nothing`, `tmerge(::PartialsLattice, ...)`
+will use the default aggressive type merge implementation that does not use `tmerge`
+recursively to accelerate convergence.
```

Comment:
> also, confusingly the tmerge function does not directly operate on the AbstractLattice subtypes, but rather the complexity lattice, and merely widens using the AbstractLattice

Could you elaborate this? Maybe you mean we want to set up another lattice for representing complexity and `tmerge` should work on it rather than the `AbstractLattice`?